### PR TITLE
I18n label-studio-frontend

### PR DESCRIPTION
**Pull request by mpomarole**
# Summary

This revision begins an internationalization effort for the frontend react app. It uses i18next and, so far, defines strings in EN and PT. 

It is not ready to be merged yet. I need input from the projects maintainers on where you want the language picker to be? And if you want the settings for language to be saved to the backend or just follow the user's current locale.

So far, I have put it in Sidebar menu, it can also be on the User's setting or footer. This is how it looks at the moment. CSS not great yet, waiting for input from you guys on where you want it before I make it pretty:

Screen Shot 2021-09-05 at 19 41 51
Screen Shot 2021-09-05 at 19 41 58

### CODE REVIEW

**Comment by nicholasrq** [comment]
whoa, that's huge! great work, thank you

our team was thinking of bringing localization to the product soon, this is a great starting point

let us review the changes, and if everything's ok, it will be merged into the main branch

File: label_studio/frontend/src/translations/pt/translations.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  timeSeries: "Série Temporal",
+  training: "Treinando",
+  uiPreview: "Visualização da UI",
+  unknown: "Desconhecido",
+  upload: "Adicionar",
+  useCloudDatabaseStorage: "Use cloud ou database storage como origem para suas tarefas de anotações ou de alvo para suas anotações completas.",
+  webhooks: "Webhooks",
+  webhooksAllowExternal: "Os webhooks permitem que serviços externos sejam notificados quando certos eventos acontecem. Quando os eventos especificados ocorrem, uma solicitação POST é enviada para cada um dos URLs fornecidos.",
+  writeInstructions: "Escreva instruções para ajudar usuários a completar tarefas de anotação.",
+  youAreAbout: "Você está prestes a deleter tudo. Essa ação não pode ser revertida.",
```
Comment (paulommaia) [inline label_studio/frontend/src/translations/pt/translations.js:130]: "apagar" / "deletar"

File: label_studio/frontend/src/translations/pt/translations.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  sourceCloudStorage: "Cloud Storage Origem",
+  startModel: "Começo o treinamento de modelo depois que anotações forem adicionadas ou atualizadas",
+  startPredfinedTemplates: "Comece com um de nossos modelos predefinidos ou crie o sua própria configuração no painel de Código",
+  startTraining: "Começar Treinamento",
+  successfullyConnected: "Conectado com sucesso",
+  syncData: "sincronizar dados de um banco de dados ou armazenamento em nuvem",
+  targetCloudStorage: "Cloud Storage Alvo",
+  taskSampling: "Amostragem de Tarefas",
+  tasks: "Tarefas",
+  tasksChosen: "As tarefas são escolhidas com aleatório uniforme",
```
Comment (paulommaia) [inline label_studio/frontend/src/translations/pt/translations.js:118]: "de forma uniforme, aleatoriamente"

File: label_studio/frontend/src/translations/pt/translations.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  saved: "Salvo",
+  seeDocumentation: "Veja a documentação para",
+  seeTheDocumentation: "veja a documentação",
+  sendForAllActions: "Enviar para todas as ações",
+  sendPayload: "Enviar Payload",
+  settings: "Configurações",
+  showBeforeLabeling: "Mostrar antes da Anotação",
+  showPredictions: "Mostrar previsões para anotadores no Label Stream e Quick View",
+  sourceCloudStorage: "Cloud Storage Origem",
+  startModel: "Começo o treinamento de modelo depois que anotações forem adicionadas ou atualizadas",
```
Comment (paulommaia) [inline label_studio/frontend/src/translations/pt/translations.js:110]: "Comece"

File: label_studio/frontend/src/translations/pt/translations.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  newWebhook: "Novo Webhook",
+  noModelVerion: "Nenhuma versão de modelo selecionado",
+  noProjects: "Não encontramos nenhum projeto",
+  optionalDescription: "Descrição opcional para seu projeto",
+  orTo: "ou para",
+  organization: "Organização",
+  payload: "Payload",
+  payloadURL: "URL da Payload",
+  performTheseActions: "Execute essas ações por sua própria conta e risco. As ações que você realiza nesta página não podem ser revertidas. Certifique-se de que seus dados tenham backup.",
+  pleaseLetUsKnow: "Please let us know in",
```
Comment (paulommaia) [inline label_studio/frontend/src/translations/pt/translations.js:88]: Falta tradução neste

File: label_studio/frontend/src/translations/pt/translations.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  labelingConfig: "A configuração de rotulagem tem mudanças essenciais que afetam a exibição de dados. Salvar a configuração pode levar à exclusão de todas as guias criadas anteriormente no Data Manager",
+  labelingConfigXML: "A configuração de anotação é basead em XML e você pode",
+  labelingEmpty: "Sua configuração de anotação está vazia. É necessário para anotar seus dados.",
+  labelingInstructions: "Instruções de Anotação",
+  labelingInterface: "Interface de Anotação",
+  labelingSetup: "Configuração de Anotação",
+  lastActivity: "Última Atividade",
+  lastActivityOn: "Última atividade no",
+  learnMore: "Saiba mais",
+  machineLearning: "Aprendizado de Máquina",
```
Comment (paulommaia) [inline label_studio/frontend/src/translations/pt/translations.js:72]: I would keep "Machine Learning" here

**Comment by mpomarole** [comment]
Obrigada @paulommaia !

**Comment by JinDing123** [comment]
Hello, how is the function of i18N going？

**Comment by mpomarole** [comment]
@JinDing123 it is very likely that I won't work on this for a while. I was waiting on feedback and response to my questions from the project maintainers before I went further with implementing the language switcher, etc. 

Let me know if more people have an interest in this, and I can consider prioritizing.

**Comment by JinDing123** [comment]
@mpomarole  
OK, I see. I personally think the I18N function should attract more users. ha ha, thank you for the label studio. It's very easy to use and provides a lot of convenience

**Comment by kevinzhao2233** [comment]
@mpomarole Hello, I think globalization is very important, this can attract many developers.

**Comment by dentalala** [comment]
I guess the language switcher can live in the sidebar now. It seems more logical to have it at user's personal page (accessible via userpic at the right side of the header) as a personal setting attached to a specific user (if it's not used too often), but we're working on new personal pages design now and things are changing fast there.

It also could use the system's current locale to determine a default language which could be changed later to other preferred language.

**Comment by mpomarole** [comment]
I won't be continuing the work here, so closing for now. 
Someone feel free to take this PR forward though with the comments on where to put the picker!

**Comment by YuantianGao** [comment]
Hi, I wanna set default language and i set the code in core, base.py:
```
# Internationalization
# https://docs.djangoproject.com/en/2.1/topics/i18n/
# LANGUAGE_CODE = 'en-us'
LANGUAGE_CODE = 'zh'
# TIME_ZONE = 'UTC'
TIME_ZONE = 'Asia/Shanghai'
# USE_I18N = False
USE_I18N = True
USE_L10N = True
USE_TZ = True
``` 
But it does not work.

**Pull request closed without merging**